models/image.py

```diff
@@ -1,10 +1,133 @@
 
-from models.base import Base
+import sqlalchemy as sa
+from sqlalchemy import orm
+from sqlalchemy.dialects.postgresql import JSONB
 
+from models.base import Base, FileOnDiskMixin, SpatiallyIndexed
+
+
+class Image(Base, FileOnDiskMixin, SpatiallyIndexed):
 
-class Image(Base):
     __tablename__ = 'images'
 
+    exposure_id = sa.Column(
+        sa.ForeignKey('exposures.id'),
+        nullable=True,
+        index=True,
+        doc=(
+            "ID of the exposure from which this image was derived. "
+            "Only set for single-image objects."
+        )
+    )
+
+    exposure = orm.relationship(
+        'Exposure',
+        cascade='save-update, merge, refresh-expire, expunge',
+        doc=(
+            "Exposure from which this image was derived. "
+            "Only set for single-image objects."
+        )
+    )
+
+    source_images = orm.relationship(
+        'Image',
+        cascade='save-update, merge, refresh-expire, expunge',
+        doc=(
+            "Images used to produce a multi-image object "
+            "(e.g., an images stack, reference, difference, super-flat, etc)."
+        )
+    )
+
+    is_multi_image = sa.Column(
+        sa.Boolean,
+        nullable=False,
+        default=False,
+        index=True,
+        doc=(
+            "Whether this image is a multi-image object. "
+            "Multi-image objects will have a null exposure, "
+            "and a list of Image objects from which it was derived. "
+        )
+    )
```

Comment:
Is this needed?  We could just check that exposure is NULL.

Or are we anticipating images that will have a NULL exposure and also not be multi-image?

What I'm worried about here is redundant information (which then can create issues for keeping the database consistent); if there are no source_images, then we know it's not a mulit-image, so we don't need a flag.

Reply:
Ok, we can use the exposure_id.